### PR TITLE
Add new mfast version

### PR DESCRIPTION
**Pull request by mathbunnyru**
Specify library name and version:  **mfast/cci.20200824**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by mathbunnyru** [comment]
Unfortunately, there were no new releases for 5 years. So I took latest commit and made a cci version.
I've asked to release a new version, but no response https://github.com/objectcomputing/mFAST/issues/94.

**Comment by mathbunnyru** [comment]
I'm not really familiar with Windows, but I think what I should do next.

1. I have to find some Windows-based image here:
https://github.com/conan-io/conan-docker-tools

2. And I have to try to compile my change using this docker image, am I right?

**Comment by mathbunnyru** [comment]
Thanks, @Croydon. I saw the error, my question was what is the easiest way to be able to reproduce the bug.
I think the best approach is to just use some VM with Windows, but I'm not sure

**Comment by mathbunnyru** [comment]
I saw you message on slack, thanks for the answer!

**Comment by mathbunnyru** [comment]
Finally, I have a windows virtual machine, where I was able to check my fix.

File: recipes/mfast/all/conanfile.py [hunk 1]
```diff
@@ -61,14 +63,13 @@ def package(self):
         cmake = self._configure_cmake()
         cmake.install()
         self.copy("licence.txt", dst="licenses", src=self._source_subfolder)
-        if self.settings.os == "Windows":
-            shutil.move(os.path.join(self.package_folder, "CMake"),
-                        os.path.join(self.package_folder, "lib", "cmake", "mFAST"))
-        os.rename(os.path.join(self.package_folder, "lib", "cmake", "mFAST", "mFASTConfig.cmake"),
-                  os.path.join(self.package_folder, "lib", "cmake", "mFAST", "mFASTTools.cmake"))
```
Comment (mathbunnyru) [inline recipes/mfast/all/conanfile.py:68]: I think it's not a good idea to move and rename these files around.
So I've just changed package_info method to choose the folder accordingly.

**Comment by Croydon** [comment]
Does the `mFASTConfig.cmake` file contain anything which can't be generated by the Conan CMake generators?

**Comment by mathbunnyru** [comment]
> Does the mFASTConfig.cmake file contain anything which can't be generated by the Conan CMake generators?

Yes, it does. Unfortunately, mfast is a cmake-only project.
The intended way to use it is to call some cmake functions and then to generate some new C++ files.

It happens in the examples here as well:
https://github.com/objectcomputing/mFAST/blob/a7c5add54acc9d73c7dff72d5a8f190fc52e9c1f/tests/CMakeLists.txt

We also do it in tests.

**Comment by mathbunnyru** [comment]
I think I made the code look quite readable.

**Comment by mathbunnyru** [comment]
Hooray, it seems the only problem left is the old compilers, so I might need to change min std version or something like that.

File: recipes/mfast/all/patches/00001-adapt-cmakelists.cci.20200824.patch.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+index ed51adc..804d77a 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -5,10 +5,13 @@
+ cmake_minimum_required(VERSION 2.8.11)
+ project (mFAST CXX)
+ 
++include(${PROJECT_BINARY_DIR}/../conanbuildinfo.cmake)
++conan_basic_setup()
++
```
Comment (prince-chrismc) [inline recipes/mfast/all/patches/00001-adapt-cmakelists.cci.20200824.patch.txt:11]: Typically we add a conan wrapper for this

File: recipes/mfast/all/patches/00001-adapt-cmakelists.cci.20200824.patch.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ # options to build tests, examples and packages
+-option(BUILD_TESTS "Build tests" ON)
+-option(BUILD_EXAMPLES "Build examples" ON)
+-option(BUILD_PACKAGES "Build packages" ON)
++option(BUILD_TESTS "Build tests" OFF)
++option(BUILD_EXAMPLES "Build examples" OFF)
++option(BUILD_PACKAGES "Build packages" OFF)
```
Comment (prince-chrismc) [inline recipes/mfast/all/patches/00001-adapt-cmakelists.cci.20200824.patch.txt:18]: These are usually configured in the recipe with `_configure_cmake()`

**Review by prince-chrismc — COMMENTED** [review]
Few things which you may look into
- compress the "link conan libs" patches into one file
- change the patch file extensionts to patch ( ie remove the `.text` )
- look into adding components, there are a lot of targets

**Comment by mathbunnyru** [comment]
Changed patch extensions.

**Comment by mathbunnyru** [comment]
Reworked patches as you suggested

**Comment by mathbunnyru** [comment]
This wasn't easy, but I think it will be much easier to update the version in the future.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by shrikantpalekar** [comment]
Hi mathbunnyru ,
    Can you please tag out a release ?

**Comment by mathbunnyru** [comment]
> Hi mathbunnyru ,
> Can you please tag out a release ?

Has something important changed since the last cci release?

**Pull request merged**